quinn-proto/src/connection/mod.rs

```diff
@@ -2311,8 +2311,7 @@ where
             } else {
                 self.config.congestion_controller_factory.build(now)
             },
-            // Try ECN on the new path if it's probably not the same as an old broken path.
-            sending_ecn: self.path.sending_ecn || !maybe_rebinding,
```

Comment:
I found the logic here quite complicated, and it made the abstraction in the next commit much harder. The downside of retrying ECN for a new path even if we think it probably won't work seems quite limited?